source/core/NumericString.cpp

```diff
@@ -1377,9 +1412,12 @@ Boolean TypedScanString(SubString* inputString, IntIndex* endToken, const Format
     
     switch (argumentType->BitEncoding()) {
         case kEncoding_UInt: 
-        case kEncoding_Enum:
             IntScanString(argument, argumentType, formatOptions->FormatChar, inpBegin, &endPointer);
             break;
+        case kEncoding_Enum:
+            if (!EnumScanString(&in, argument, argumentType, formatOptions->FormatChar, inpBegin, &endPointer))
```

Comment:
Interesting, in general I prefer to have a single method that takes care of all the conditions, but I can see your point here, given that it will shown more clearly what is going on in this case structure. I'll consider changing the code in a future commit.